engine/sources/gitdns/source.go

```diff
@@ -247,10 +248,21 @@ func (gs *gitSourceHandler) getAuthToken(ctx context.Context, g session.Group) e
 				}
 				return err
 			}
-			if s.token {
-				dt = []byte("basic " + base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("x-access-token:%s", dt))))
+
+			if strings.Contains(gs.src.Remote, "bitbucket.org") {
+				// For Bitbucket Cloud Git operations, always use x-token-auth
+				// https://support.atlassian.com/bitbucket-cloud/docs/using-access-tokens/
+				gs.auth = []string{
+					"-c", fmt.Sprintf("credential.helper=!f() { echo \"username=x-token-auth\"; echo \"password=%s\"; }; f", string(dt)),
+					"-c", "credential.https://bitbucket.org.username=x-token-auth",
+					"-c", "credential.useHttpPath=true",
+				}
```

Comment:
suggestion: I would apply the logic for detecting this in `authSecretNames`, and then extend `authSecret` to have a `credhelper bool` property. This keeps the logic decoupled, but also means that we don't end up trying to put `AuthHeaderSecrets`s as passwords (we should only put the tokens there).

If this is called using an `AuthHeader`, we **shouldn't** be plugging the entire auth header in as a `password` - we should plug it in as a header as requested. It's only for the `AuthToken` that we want to apply the logic in this PR.

Reply:
agreed, updated 🙏